blockstack/stacks-blockchain-sidecar
# Contract-call tx broadcast testing endpoint and Clarity ABI parsing

Development / testing endpoint for performing a contract-call tx. 

### Testing

1. Pull core-node branch `genesis-contract-abi` https://github.com/blockstack/stacks-blockchain/tree/feature/genesis-contract-abi and configure the Stacks.toml file to match sidecar address and port.
2. Deploy a contract using page http://127.0.0.1:3999/debug/broadcast/contract-deploy
![image](https://user-images.githubusercontent.com/1447546/78241877-60e05680-74e1-11ea-9945-277c18035d03.png)
3. Navigate to the deployed contract page http://127.0.0.1:3999/debug/broadcast/contract-call/ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH.kv-store
![image](https://user-images.githubusercontent.com/1447546/78241998-94bb7c00-74e1-11ea-80a9-7957459b32e2.png)
4. Submit a contract call and navigate to the tx result page to see API result: http://127.0.0.1:3999/tx/0x13370d7e9996b724f3def76809c8bfc15c0bc53e612ba32b01d44d49f77c958c
```json
{
  "block_hash": "0xd3939df3599fce22a37d477bd86040df30e5ed0fb051648549286f137bcf4146",
  "block_height": 48,
  "tx_id": "0x13370d7e9996b724f3def76809c8bfc15c0bc53e612ba32b01d44d49f77c958c",
  "tx_index": 1,
  "tx_status": "success",
  "tx_type": "contract_call",
  "fee_rate": "9",
  "sender_address": "ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH",
  "sponsored": false,
  "contract_call": {
    "contract_id": "ST2ZRX0K27GW0SP3GJCEMHD95TQGJMKB7G9Y0X1MH.kv-store",
    "function_name": "set-value",
    "function_args": [
      "0x020000000568656c6c6f",
      "0x0200000005776f726c64"
    ]
  },
  "events": []
}
```
